### PR TITLE
Cleanup old traversal code 

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

EE PR: https://github.com/arangodb/enterprise/pull/935

This PR removes a lot of code which will be obsolete and replaced with the graph refactored code components.

- [x] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [x] Backports: No backporst needed

#### Related Information

- [x] Enterprise PR:

### CODE REVIEW

**Comment by hkernbach** [comment]
Jenkins green.

File: tests/Aql/TraversalExecutorTest.cpp [hunk 1]
```diff
@@ -1,715 +0,0 @@
...
-/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-/// See the License for the specific language governing permissions and
-/// limitations under the License.
-///
-/// Copyright holder is ArangoDB GmbH, Cologne, Germany
-///
-/// @author Michael Hackstein
-////////////////////////////////////////////////////////////////////////////////
-
-#include "gtest/gtest.h"
```
Comment (mchacki) [inline tests/Aql/TraversalExecutorTest.cpp:24]: Why are we removing this file?
As far as I see it should be testing the AQL ExecutionNode.
Should we not move this to use a MockGraph / MockGraphProvider?
It should also be able to use a SingleServerTraversal here.

**Review by mchacki — CHANGES_REQUESTED** [review]
We should quickly discuss the test, and how much effort I would be to retain it, it shall be testing the refactored variant now.
But I am not too bound to this test.

File: tests/Aql/TraversalExecutorTest.cpp (hunk 1 above)
Comment (mchacki) [reply tests/Aql/TraversalExecutorTest.cpp:24]: Discussed to delete it for now, and potentially revive it later on.
We kept it in Kansan history.
The items under test here are covered by Integration tests anyways.

File: arangod/Aql/TraversalNode.cpp [hunk 2]
```diff
@@ -1368,7 +1142,7 @@ void TraversalNode::prepareOptions() {
   TraverserOptions* opts = this->TraversalNode::options();
   TRI_ASSERT(opts != nullptr);
   /*
-   * HACK: DO NOT use other indexes for smart BFS. Otherwise this will produce
+   * HACK: DO NOT use other indexes for smart BFS. Otherwise, this will produce
```
[the remark this answers is not in the document]
Comment (hkernbach) [reply arangod/Aql/TraversalNode.cpp:1145]: Created card

File: tests/Aql/TraversalExecutorTest.cpp (hunk 1 above)
Comment (hkernbach) [reply tests/Aql/TraversalExecutorTest.cpp:24]: Created card

**Pull request merged**